[core] "Unknown option: '-min'"  but is referenced in documentation

<!-- Please, prefix the report title with the language it applies to within brackets, such as [java] or [apex].
If not specific to a language, you can use [core]. -->

**Affects PMD Version:** 

7.0.0-rc1

**Description:**

The `-min` parameter is no longer allowed. I cannot use the priority (1-5) for rules to exclude some from being checked.

`--minimum priority` is a Low/Medium/High value, so that can not be the solution:
```
Rule priority threshold; rules with lower priority than configured here won't be used. Valid values (case insensitive): High, Medium High, Medium, Medium Low, Low
```

Substituting --minimum-priority gives a long error message basically saying that it is not a number, but a string.

[-min is referenced in documentation](https://docs.pmd-code.org/latest/pmd_userdocs_configuring_rules.html#message-and-priority-overriding)


```
Run ~/pmd/bin/pmd check -d force-app -d sample-handler --rulesets pmd/ruleset.xml --format csv --no-cache -min 2 --verbose -r pmd/pmd.csv
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.codehaus.groovy.reflection.CachedClass (file:/home/runner/pmd/lib/groovy-2.[4](https://github.com/dschach/salesforce-trigger-framework/actions/runs/4622271015/jobs/8174740004#step:4:5).21.jar) to method java.lang.Object.finalize()
WARNING: Please consider reporting this to the maintainers of org.codehaus.groovy.reflection.CachedClass
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
Unknown option: '-min'
Possible solutions: --minimum-priority
Error: Process completed with exit code 2.
```

**Code Sample demonstrating the issue:**

```bash
~/pmd/bin/pmd check --dir force-app --dir sample-handler --rulesets pmd/ruleset.xml --format csv --no-cache -min 2 --verbose -r pmd/pmd.csv
```

**Steps to reproduce:**

Please provide detailed steps for how we can reproduce the bug.

Run pmd with `-min` parameter and 

**Running PMD through:** *GitHub Action*: https://github.com/dschach/salesforce-trigger-framework/blob/main/.github/workflows/ci.yml

<!-- If relevant, also include your JDK and OS information, e.g. for ClassNotFoundException, LinkageError, reflection failures, etc. -->

--- a/docs/pages/pmd/userdocs/cli_reference.md
+++ b/docs/pages/pmd/userdocs/cli_reference.md
@@ -127,10 +127,10 @@ The tool comes with a rather extensive help text, simply running with `--help`!
     %}
     {% include custom/cli_option_row.html options="--minimum-priority"
                option_arg="priority"
-               description="Rule priority threshold; rules with lower priority
-                              than configured here won't be used.
-                            Valid values (case insensitive): High, Medium High,
-                              Medium, Medium Low, Low"
+               description="Rule priority threshold; rules with lower priority than configured here won't be used.
+                            Valid values (case insensitive): High, Medium_High, Medium, Medium_Low, Low.
+                            An integer between 1 (High) and 5 (Low) is also supported. See [Configuring rules](pmd_userdocs_configuring_rules.html)
+                            on how to override priorities in custom rulesets."
                default="Low"
     %}
     {% include custom/cli_option_row.html options="--no-ruleset-compatibility"

--- a/docs/pages/pmd/userdocs/configuring_rules.md
+++ b/docs/pages/pmd/userdocs/configuring_rules.md
@@ -4,7 +4,7 @@ short_title: Configuring rules
 keywords: [property, properties, message, priority]
 tags: [userdocs, getting_started]
 summary: "Learn how to configure your rules directly from the ruleset XML."
-last_updated: May 2018 (6.4.0)
+last_updated: May 2023 (7.0.0)
 permalink: pmd_userdocs_configuring_rules.html
 author: Hooper Bloob <hooperbloob@users.sourceforge.net>, Romain Pelisse <rpelisse@users.sourceforge.net>, Clément Fournier <clement.fournier76@gmail.com>
 ---
@@ -17,13 +17,20 @@ value and change the message the rule will print on the report.
 
 Similarly, the **priority** of a rule can be changed via a nested
 element. Using priority, you can deactivate some rules based on a
-minimum priority threshold (set using the `-min` CLI option).
-Priority is an integer ranging from 1 to 5, with 1 being the highest
-priority.
+minimum priority threshold (set using the `--minimum-priority` CLI option).
+Priority in the ruleset is an integer ranging from 1 to 5, with 1 being the highest
+priority. On the command line, you need to use the following values:
 
+| Priority | --minimum-priority value |
+|----------|--------------------------|
+| 1        | High                     |
+| 2        | Medium High              |
+| 3        | Medium                   |
+| 4        | Medium Low               |
+| 5        | Low                      |
 
 Putting things together, the following rule reference lowers the priority
-of EmptyCatchBlock to 5, such that e.g. using the `-min 4` CLI parameters
+of EmptyCatchBlock to 5, such that e.g. using the `--minimum-priority=Medium` CLI parameters
 will cause the rule to be ignored.
 
 ```xml

--- a/docs/pages/pmd/userdocs/configuring_rules.md
+++ b/docs/pages/pmd/userdocs/configuring_rules.md
@@ -19,18 +19,18 @@ Similarly, the **priority** of a rule can be changed via a nested
 element. Using priority, you can deactivate some rules based on a
 minimum priority threshold (set using the `--minimum-priority` CLI option).
 Priority in the ruleset is an integer ranging from 1 to 5, with 1 being the highest
-priority. On the command line, you need to use the following values:
+priority. On the command line, you can either use the integer or the following values:
 
 | Priority | --minimum-priority value |
 |----------|--------------------------|
 | 1        | High                     |
-| 2        | Medium High              |
+| 2        | Medium_High              |
 | 3        | Medium                   |
-| 4        | Medium Low               |
+| 4        | Medium_Low               |
 | 5        | Low                      |
 
 Putting things together, the following rule reference lowers the priority
-of EmptyCatchBlock to 5, such that e.g. using the `--minimum-priority=Medium` CLI parameters
+of EmptyCatchBlock to 5 ("Low"), such that e.g. using the `--minimum-priority=Medium_Low` CLI parameters
 will cause the rule to be ignored.
 
 ```xml

--- a/docs/pages/pmd/userdocs/extending/rule_guidelines.md
+++ b/docs/pages/pmd/userdocs/extending/rule_guidelines.md
@@ -2,7 +2,7 @@
 title: Rule guidelines
 tags: [extending, userdocs]
 summary: "Rule Guidelines, or the last touches to a rule"
-last_updated: February 2020 (6.22.0)
+last_updated: Mai 2023 (7.0.0)
 permalink: pmd_userdocs_extending_rule_guidelines.html
 author: Xavier Le Vourch, Ryan Gustafson, Romain Pelisse
 ---
@@ -12,26 +12,37 @@ Here is a bunch of thing to do you may consider once your rule is “up and runn
 
 ## How to define rules priority
 
-Rule priority may, of course, changes a lot depending on the context of the project. However, you can use the following guidelines to assert the legitimate priority of your rule:
+Rule priority may, of course, changes a lot depending on the context of the project. However, you can use the
+following guidelines to assert the legitimate priority of your rule:
 
-1.  **Change absolutely required.** Behavior is critically broken/buggy.
-2.  **Change highly recommended.** Behavior is quite likely to be broken/buggy.
-3.  **Change recommended.** Behavior is confusing, perhaps buggy, and/or against standards/best practices.
-4.  **Change optional.** Behavior is not likely to be buggy, but more just flies in the face of standards/style/good taste.
-5.  **Change highly optional.** Nice to have, such as a consistent naming policy for package/class/fields…
+1.  **High: Change absolutely required.** Behavior is critically broken/buggy.
+2.  **Medium High: Change highly recommended.** Behavior is quite likely to be broken/buggy.
+3.  **Medium: Change recommended.** Behavior is confusing, perhaps buggy, and/or against standards/best practices.
+4.  **Medium Low: Change optional.** Behavior is not likely to be buggy, but more just flies in the face of
+    standards/style/good taste.
+5.  **Low: Change highly optional.** Nice to have, such as a consistent naming policy for package/class/fields…
 
-For instance, let’s take the ExplicitCallToGC rule (“Do not explicitly trigger a garbage collection.”). Calling GC is a bad idea, but it doesn’t break the application. So we skip priority one. However, as explicit call to gc may really hinder application performances, we set for the priority 2.
+For instance, let's take the ExplicitCallToGC rule (“Do not explicitly trigger a garbage collection.”). Calling GC is
+a bad idea, but it doesn't break the application. So we skip priority one. However, as explicit call to gc may really
+hinder application performances, we set for the priority 2 ("Medium High").
 
 ## Correctness
 
-You should try to run the rule on a large code base, like the jdk source code for instance. This will help ensure that the rule does not raise exceptions when dealing with unusual constructs.
+You should try to run the rule on a large code base, like the jdk source code for instance. This will help ensure
+that the rule does not raise exceptions when dealing with unusual constructs.
 
-If your rule is stateful, make sure that it is reinitialized correctly. The “-stress” command line option can be used as the files will then not be ordered but processed randomly. Running pmd with the “-stress” option several times and sorting the text output should produce identical results if the state information is correctly reset.
+If your rule is stateful, make sure that it is reinitialized correctly for each file.
 
 ## Performance issues
 
-When writing a new rule, using command line option “-benchmark” on a few rules can give an indication on how the rule compares to others. To get the full picture, use the rulesets/internal/all-java.xml ruleset with “-benchmark”.
+When writing a new rule, using command line option `--benchmark` on a few rules can give an indication on how
+the rule compares to others. To get the full picture, use the `rulesets/internal/all-java.xml` ruleset
+with “--benchmark”.
 
-Rules which use the RuleChain to visit the AST are faster than rules which perform manual visitation of the AST. The difference is small for an individual Java rule, but when running 100s of rules, it is measurable. For XPath rules, the difference is extremely noticeable due to Jaxen overhead for AST navigation. Make sure your XPath rules using the RuleChain.
+Rules which use the RuleChain to visit the AST are faster than rules which perform manual visitation of the AST.
+The difference is small for an individual Java rule, but when running 100s of rules, it is measurable.
+For XPath rules, the difference is extremely noticeable due to the overhead for AST navigation.
+Make sure your XPath rules using the RuleChain. If RuleChain can't be used for your XPath rule, then this fact
+is logged as a debug message.
 
 (TODO How does one know except by running in a debugger or horrendous performance?).

--- a/docs/pages/release_notes.md
+++ b/docs/pages/release_notes.md
@@ -41,6 +41,8 @@ The remaining section describes the complete release notes for 7.0.0.
 
 #### Fixed issues
 
+* core
+  * [#4454](https://github.com/pmd/pmd/issues/4454): \[core] "Unknown option: '-min'" but is referenced in documentation
 * java-errorprone
   * [#4457](https://github.com/pmd/pmd/issues/4457): \[java] OverrideBothEqualsAndHashcode: false negative with anonymous classes
   * [#4546](https://github.com/pmd/pmd/issues/4546): \[java] OverrideBothEqualsAndHashCode ignores records
@@ -53,6 +55,10 @@ The remaining section describes the complete release notes for 7.0.0.
     * `net.sourceforge.pmd.cli.PMDCommandLineInterface`
     * `net.sourceforge.pmd.cli.PMDParameters`
     * `net.sourceforge.pmd.cli.PmdParametersParseResult`
+
+* The CLI option `--stress` (or `-stress`) has been removed without replacement.
+* The CLI option `--minimum-priority` now takes one of the following values instead of an integer:
+  High, Medium High, Medium, Medium Low, Low.
 
 #### External Contributions
 
@@ -359,6 +365,7 @@ See [Detailed Release Notes for PMD 7]({{ baseurl }}pmd_release_notes_pmd7.html)
     * [#4353](https://github.com/pmd/pmd/pull/4353):   \[core] Micro optimizations for Node API
     * [#4365](https://github.com/pmd/pmd/pull/4365):   \[core] Improve benchmarking
     * [#4420](https://github.com/pmd/pmd/pull/4420):   \[core] Remove PMD.EOL
+    * [#4454](https://github.com/pmd/pmd/issues/4454): \[core] "Unknown option: '-min'" but is referenced in documentation
 * cli
     * [#2234](https://github.com/pmd/pmd/issues/2234): \[core] Consolidate PMD CLI into a single command
     * [#3828](https://github.com/pmd/pmd/issues/3828): \[core] Progress reporting

--- a/docs/pages/release_notes.md
+++ b/docs/pages/release_notes.md
@@ -57,8 +57,9 @@ The remaining section describes the complete release notes for 7.0.0.
     * `net.sourceforge.pmd.cli.PmdParametersParseResult`
 
 * The CLI option `--stress` (or `-stress`) has been removed without replacement.
-* The CLI option `--minimum-priority` now takes one of the following values instead of an integer:
-  High, Medium High, Medium, Medium Low, Low.
+* The CLI option `--minimum-priority` was changed with 7.0.0-rc1 to only take the following values:
+  High, Medium High, Medium, Medium Low, Low. With 7.0.0-rc2 compatibility has been restored, so that the equivalent
+  integer values (1 to 5) are supported as well.
 
 #### External Contributions
 

--- a/docs/pages/release_notes_pmd7.md
+++ b/docs/pages/release_notes_pmd7.md
@@ -967,6 +967,10 @@ Related issue: [[core] Language lifecycle (#3782)](https://github.com/pmd/pmd/is
     * `net.sourceforge.pmd.cli.PMDParameters`
     * `net.sourceforge.pmd.cli.PmdParametersParseResult`
 
+* The CLI option `--minimum-priority` was changed with 7.0.0-rc1 to only take the following values:
+  High, Medium High, Medium, Medium Low, Low. With 7.0.0-rc2 compatibility has been restored, so that the equivalent
+  integer values (1 to 5) are supported as well.
+
 #### 7.0.0-rc1
 
 * The CLI option `--stress` (or `-stress`) has been removed without replacement.

--- a/docs/pages/release_notes_pmd7.md
+++ b/docs/pages/release_notes_pmd7.md
@@ -378,7 +378,7 @@ Commands:
               Warning: May not support the full CPD feature set
   ast-dump  Experimental: dumps the AST of parsing source code
 Exit Codes:
-  0   Succesful analysis, no violations found
+  0   Successful analysis, no violations found
   1   An unexpected error occurred during execution
   2   Usage error, please refer to the command help
   4   Successful analysis, at least 1 violation found
@@ -966,6 +966,12 @@ Related issue: [[core] Language lifecycle (#3782)](https://github.com/pmd/pmd/is
     * `net.sourceforge.pmd.cli.PMDCommandLineInterface`
     * `net.sourceforge.pmd.cli.PMDParameters`
     * `net.sourceforge.pmd.cli.PmdParametersParseResult`
+
+#### 7.0.0-rc1
+
+* The CLI option `--stress` (or `-stress`) has been removed without replacement.
+* The CLI option `--minimum-priority` now takes one of the following values instead of an integer:
+  High, Medium High, Medium, Medium Low, Low.
 
 #### 6.55.0
 

--- a/pmd-cli/src/main/java/net/sourceforge/pmd/cli/commands/internal/PmdCommand.java
+++ b/pmd-cli/src/main/java/net/sourceforge/pmd/cli/commands/internal/PmdCommand.java
@@ -27,6 +27,7 @@ import net.sourceforge.pmd.benchmark.TimingReport;
 import net.sourceforge.pmd.benchmark.TimingReportRenderer;
 import net.sourceforge.pmd.cli.commands.typesupport.internal.PmdLanguageTypeSupport;
 import net.sourceforge.pmd.cli.commands.typesupport.internal.PmdLanguageVersionTypeSupport;
+import net.sourceforge.pmd.cli.commands.typesupport.internal.RulePriorityTypeSupport;
 import net.sourceforge.pmd.cli.internal.CliExitCode;
 import net.sourceforge.pmd.cli.internal.ProgressBarListener;
 import net.sourceforge.pmd.internal.LogMessages;
@@ -173,7 +174,8 @@ public class PmdCommand extends AbstractAnalysisPmdSubcommand {
     @Option(names = "--minimum-priority",
             description = "Rule priority threshold; rules with lower priority than configured here won't be used.%n"
                     + "Valid values (case insensitive): ${COMPLETION-CANDIDATES}",
-            defaultValue = "Low")
+            defaultValue = "Low",
+            completionCandidates = RulePriorityTypeSupport.class, converter = RulePriorityTypeSupport.class)
     public void setMinimumPriority(final RulePriority priority) {
         this.minimumPriority = priority;
     }

--- a/pmd-cli/src/main/java/net/sourceforge/pmd/cli/commands/typesupport/internal/RulePriorityTypeSupport.java
+++ b/pmd-cli/src/main/java/net/sourceforge/pmd/cli/commands/typesupport/internal/RulePriorityTypeSupport.java
@@ -1,0 +1,38 @@
+/*
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.cli.commands.typesupport.internal;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import net.sourceforge.pmd.RulePriority;
+
+import picocli.CommandLine.ITypeConverter;
+import picocli.CommandLine.TypeConversionException;
+
+public class RulePriorityTypeSupport implements ITypeConverter<RulePriority>, Iterable<String> {
+    @Override
+    public RulePriority convert(String value) {
+        for (RulePriority rulePriority : RulePriority.values()) {
+            String descriptiveName = rulePriority.getName();
+            String name = rulePriority.name();
+            String priority = String.valueOf(rulePriority.getPriority());
+            if (descriptiveName.equalsIgnoreCase(value) || name.equalsIgnoreCase(value) || priority.equalsIgnoreCase(value)) {
+                return rulePriority;
+            }
+        }
+        throw new TypeConversionException("Invalid priority: " + value);
+    }
+
+    @Override
+    public Iterator<String> iterator() {
+        List<String> descriptiveNames = new ArrayList<>();
+        for (RulePriority rulePriority : RulePriority.values()) {
+            descriptiveNames.add(rulePriority.getName());
+        }
+        return descriptiveNames.iterator();
+    }
+}

--- a/pmd-cli/src/main/java/net/sourceforge/pmd/cli/commands/typesupport/internal/RulePriorityTypeSupport.java
+++ b/pmd-cli/src/main/java/net/sourceforge/pmd/cli/commands/typesupport/internal/RulePriorityTypeSupport.java
@@ -29,10 +29,10 @@ public class RulePriorityTypeSupport implements ITypeConverter<RulePriority>, It
 
     @Override
     public Iterator<String> iterator() {
-        List<String> descriptiveNames = new ArrayList<>();
+        List<String> completionValues = new ArrayList<>();
         for (RulePriority rulePriority : RulePriority.values()) {
-            descriptiveNames.add(rulePriority.getName());
+            completionValues.add(rulePriority.name());
         }
-        return descriptiveNames.iterator();
+        return completionValues.iterator();
     }
 }

--- a/pmd-cli/src/test/java/net/sourceforge/pmd/cli/PmdCliTest.java
+++ b/pmd-cli/src/test/java/net/sourceforge/pmd/cli/PmdCliTest.java
@@ -485,6 +485,16 @@ class PmdCliTest extends BaseCliTest {
         runCliSuccessfully("-d", srcDir.toString(), "-f", "text", "-R", RULESET_WITH_VIOLATION, "--minimum-priority", "Medium_High");
     }
 
+    @Test
+    void minimumPriorityOptionNumeric() throws Exception {
+        runCli(VIOLATIONS_FOUND, "-d", srcDir.toString(), "-f", "text", "-R", RULESET_WITH_VIOLATION, "--minimum-priority", "3")
+                .verify(r -> r.checkStdOut(
+                        containsString("Violation from ReportAllRootNodes")
+                ));
+        runCliSuccessfully("-d", srcDir.toString(), "-f", "text", "-R", RULESET_WITH_VIOLATION, "--minimum-priority", "1");
+        runCliSuccessfully("-d", srcDir.toString(), "-f", "text", "-R", RULESET_WITH_VIOLATION, "--minimum-priority", "2");
+    }
+
     // utilities
     private Path tempRoot() {
         return tempDir;

--- a/pmd-cli/src/test/java/net/sourceforge/pmd/cli/PmdCliTest.java
+++ b/pmd-cli/src/test/java/net/sourceforge/pmd/cli/PmdCliTest.java
@@ -474,6 +474,17 @@ class PmdCliTest extends BaseCliTest {
                 ));
     }
 
+    @Test
+    void minimumPriorityOption() throws Exception {
+        runCli(VIOLATIONS_FOUND, "-d", srcDir.toString(), "-f", "text", "-R", RULESET_WITH_VIOLATION, "--minimum-priority", "Medium")
+                .verify(r -> r.checkStdOut(
+                        containsString("Violation from ReportAllRootNodes")
+                ));
+        runCliSuccessfully("-d", srcDir.toString(), "-f", "text", "-R", RULESET_WITH_VIOLATION, "--minimum-priority", "High");
+        runCliSuccessfully("-d", srcDir.toString(), "-f", "text", "-R", RULESET_WITH_VIOLATION, "--minimum-priority", "medium HIGH");
+        runCliSuccessfully("-d", srcDir.toString(), "-f", "text", "-R", RULESET_WITH_VIOLATION, "--minimum-priority", "Medium_High");
+    }
+
     // utilities
     private Path tempRoot() {
         return tempDir;

--- a/pmd-core/src/main/java/net/sourceforge/pmd/RulePriority.java
+++ b/pmd-core/src/main/java/net/sourceforge/pmd/RulePriority.java
@@ -12,7 +12,7 @@ package net.sourceforge.pmd;
  * avoided in favor of {@link RulePriority#getPriority()} and
  * {@link RulePriority#valueOf(int)}
  *
- * @see <a href="http://pmd.sourceforge.net/pmd-5.1.2/rule-guidelines.html">How
+ * @see <a href="https://docs.pmd-code.org/latest/pmd_userdocs_extending_rule_guidelines.html">How
  *      to define rules priority</a>
  */
 public enum RulePriority {

